spyder/widgets/editor.py

```diff
@@ -274,6 +274,70 @@ def breakpoints_changed(self):
             self.save_breakpoints.emit(self.filename, repr(breakpoints))
 
 
+class TabSwitcherWidget(QListWidget):
+    """Show tabs in mru order and change between them."""
+    def __init__(self, parent, stack_history, tabs):
+        QListWidget.__init__(self, parent)
+        self.setWindowFlags(Qt.SubWindow | Qt.FramelessWindowHint)
+
+        self.editor = parent
+        self.stack_history = stack_history
+        self.tabs = tabs
+
+        self.setSelectionMode(QListWidget.SingleSelection)
+        self.itemActivated.connect(self.item_selected)
+
+        self.id_list = []
+        self.load_data()
+        size = CONF.get('main', 'completion/size')

```

Comment:
I think it's better to have both widgets (the file switcher and this one) placed at the same location to avoid breaking expectations from users (i.e. where to expect each dialog in the screen). Since the file switcher came first, this one has to follow it.